tools/Az.Tools.Predictor/Az.Tools.Predictor.SourceGenerator/SettingsSourceGenerator.cs

```diff
@@ -0,0 +1,199 @@
+
+    public string Value { get; set; }
+}
+
+/// <summary>
+/// The source generator to create the part of the Settings class from the json file.
+///
+/// It gathers the key:value pair from the json file, and generates the public properties accordingly.
+/// The key from the json is the property name and the value from the json is the property value. It infers
+/// the property type based on the value.
+/// The generated properties are the default values of the settings.
+/// In addition to generating the properties, it also generate the method to get the settings from the user's profile,
+/// and override the property accordingly if it's set in the user's profile setting.
+/// </summary>
+[Generator]
+public class SettingsSourceGenerator : ISourceGenerator
+{
+    private const string _StringType = "string";
+    private const string _IntType = "int?";
+    private const string _DoubleType = "double?";
+    private const string _BoolType = "bool?";
+
+    /// <inheritdoc />
+    public void Execute(GeneratorExecutionContext context)
+    {
+        var settingClass = context.Compilation.GetTypeByMetadataName("Microsoft.Azure.PowerShell.Tools.AzPredictor.Settings");
+
+        if (settingClass is null)
+        {
+            throw new InvalidOperationException("Is the class Settings renamed?");
+        }
+
+        var generatedSource = new StringBuilder($@"
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the ""License"");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an ""AS IS"" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace {settingClass.ContainingNamespace}
+{{
+    // This file is generated from the source generator. DO NOT edit this file.
+    sealed partial class {settingClass.Name}
+    {{
+");
+
+        var settingFilePath = context.AdditionalFiles.Single((file) => file.Path.EndsWith("AzPredictorSettings.json", StringComparison.Ordinal)).Path;
+
+        var fileContent = File.ReadAllText(settingFilePath, Encoding.UTF8);
+        var jsonDocument = JsonDocument.Parse(fileContent);
+        var jsonRoot = jsonDocument.RootElement;
+        var settingProperties = new List<SettingProperty>();
+
+        // Parse the json file and gather the property information, including name, type, and value.
+
+        foreach (var element in jsonRoot.EnumerateObject())
+        {
+            var addedProperty = new SettingProperty()
```

Comment:
nit: maybe something like "newProperty" or "generatedProperty"? I'm okay if it stays as addedProperty, just found myself reading this as new or generated property.